PTHREAD_STACK_MIN is no longer a compile time define in glibc 2.34+ and since
we only care for glibc and musl where PTHREAD_STACK_MIN is always defined there
is no need to check for constant called PTHREAD_STACK_MIN since its already defined
this fix may not work for wider audience but for OE needs its sufficient

Upstream-Status: Inappropriate [OE-only fix]
Signed-off-by: Khem Raj <raj.khem@gmail.com>
--- a/src/third_party/boost-1.70.0/boost/thread/pthread/thread_data.hpp
+++ b/src/third_party/boost-1.70.0/boost/thread/pthread/thread_data.hpp
@@ -57,9 +57,7 @@ namespace boost
 #else
           std::size_t page_size = ::sysconf( _SC_PAGESIZE);
 #endif
-#if PTHREAD_STACK_MIN > 0
           if (size<PTHREAD_STACK_MIN) size=PTHREAD_STACK_MIN;
-#endif
           size = ((size+page_size-1)/page_size)*page_size;
           int res = pthread_attr_setstacksize(&val_, size);
           BOOST_VERIFY(!res && "pthread_attr_setstacksize failed");
